backend/parser.py

```diff
@@ -124,17 +131,21 @@ def __init__(self, files: list, regex: dict):
         """
 
         self.files = files
-        self.regex = regex
+        self.compiled = {}
         self.samples = []
 
         try: # Test if any required fields are not defined.
```

Comment:
I should have noticed this before, but it would be better to check if the key is in the key list as opposed to seeing if you throw an exception. Then you could raise the KeyError with the particular missing field.